apps/service_broadcast/lib/broadcast/stream/store.ex

```diff
@@ -35,9 +35,15 @@ defmodule Broadcast.Stream.Store do
     Brook.ViewState.delete(@collection, identifier(dataset_id, subset_id))
   end
 
-  @spec get_all!() :: [Load.Broadcast.t()]
-  def get_all!() do
-    Brook.get_all_values!(@instance, @collection)
-    |> Enum.map(&Map.get(&1, "load"))
+  @spec get_all() :: [Load.t()]
+  def get_all() do
+    case Brook.get_all_values(@instance, @collection) do
+      {:ok, results} ->
+        results |> Enum.map(&Map.get(&1, "load"))
+        {:ok, results}
+
+      {:error, _} ->
+        {:error, "Failed to get values from Broadcast store"}
+    end
```

Comment:
consider changing this to a `with` without the else block and pipe the result to the `Ok.map/2` function in the umbrella's Ok library so you can just pass the `{:ok, result}` straight to to Map like:

```suggestion
    with {:ok, _state} = results <- Brook.get_all_values(@instance, @collection) do
      Ok.map(results, &Map.get(&1, "load"))
    end
```
Plus, the way this is currently written, I don't think you're returning the thing you want since you pipe the results to the `Enum.map` call but then you don't re-bind results so you're just returning the results you go straight out of Brook pre-Map.

Reply:
Do I need an else statement or are we good just letting this bubble up an error from the underlying call to Brook.get_all_values?